src/Okta.Sdk/Internal/DefaultDataStore.cs

```diff
@@ -222,8 +227,8 @@ public async Task<HttpResponse<TResponse>> PostAsync<TResponse>(HttpRequest requ
             PrepareRequest(request, requestContext);
 
             var body = _serializer.Serialize(request.Payload);
-
-            var response = await _requestExecutor.PostAsync(request.Uri, request.Headers, body, cancellationToken).ConfigureAwait(false);
+            request.SetBody(body);
```

Comment:
Since we have a `PreparaRequest` method, should we move this code there?

Reply:
Setting the body in this way is only appropriate here due to the way that the body/payload is handled in the broader context.